heiwa/assets/theme.css

```diff
@@ -0,0 +1,28 @@
+/**
+ * It looks like the responsive navigation is forced to be the wide width. https://github.com/WordPress/gutenberg/pull/43576
+ */
+
+.wp-block-navigation__responsive-close {
+	max-width: none;
+}
+
+/**
+ * Override the default padding value from Gutenberg to avoid the icon jumping
+ */
+.wp-block-navigation__responsive-container.is-menu-open {
+	padding-top: var(--wp--custom--gap--vertical);
+}
+
+/**
+ * Override Blockbase ponyfill style that makes navigation items left aligned.
```

Comment:
It seems to be an issue with all blockbase children, and having a fix in blockbase would be more appropriate.

The following should work without breaking child themes.

`_navigation.scss`, Lines 32 and 39
```
align-items: var(--navigation-layout-justification-setting, flex-start);
```